File: src/test/scala/org/apache/flume/source/ftp/ExternalFtpSourceTests.scala
package org.apache.flume.source.ftp

import java.nio.file.{Files, FileSystems}

import org.apache.flume.PollableSource
import org.apache.flume.source.ftp.server.EmbeddedFTPServer
import org.apache.flume.source.{TestFileUtils, TestFileUtilsTest}
import org.testng.Assert._
import org.testng.annotations.Test

/**
 * Integration tests that will connect to an external FTP server.
 *
 * Created by luca on 4/2/15.
 */
class ExternalFtpSourceTests extends AbstractFtpSourceTest with TestFileUtils{
  override def getPort: Int = 21
  override def getPassword: String = "anonymous"
  override def getUser: String = "anonymous"

  @Test(enabled = false)
  def symLinkTest(): Unit ={
    val proc0 = ftpSource.process
    assertEquals(PollableSource.Status.READY, proc0)
    assertEquals(ftpSourceCounter.getFilesCount,1)
    assertEquals(ftpSourceCounter.getFilesProcCount,1)
    assertEquals(ftpSourceCounter.getFilesProcCountError,0)

  }

}

File: src/test/java/org/apache/flume/source/ftp/ExternalFtpSourceTests.java
//package org.apache.flume.source.ftp;
//
//import org.apache.flume.EventDeliveryException;
//import org.apache.flume.PollableSource;
//import org.testng.Assert;
//import org.testng.annotations.Test;
//
///**
// * Integration tests that will connect to an external FTP server.
// *
// */
//public class ExternalFtpSourceTests extends AbstractFtpSourceTest {
//    int getPort = 21;
//    String getPassword = "anonymous";
//    String getUser = "anonymous";
//
//    @Test(enabled = false)
//    public void symLinkTest() {
//        PollableSource.Status proc0 = null;
//        try {
//            proc0 = ftpSource.process();
//            Assert.assertEquals(PollableSource.Status.READY, proc0);
//            Assert.assertEquals(ftpSourceCounter.getFilesCount(), 1);
//            Assert.assertEquals(ftpSourceCounter.getFilesProcCount(), 1);
//            Assert.assertEquals(ftpSourceCounter.getFilesProcCountError(), 0);
//        } catch (EventDeliveryException e) {
//            Assert.fail();
//        }
//    }
//
//}
